core/variant/type_info.h

```diff
@@ -367,4 +367,82 @@ ZERO_INITIALIZER_NUMBER(char32_t)
 ZERO_INITIALIZER_NUMBER(float)
 ZERO_INITIALIZER_NUMBER(double)
 
+template <typename T>
+inline constexpr Variant::Type get_variant_type_v = TypeInherits<Object, T>::value ? Variant::OBJECT : Variant::VARIANT_MAX;
+
+template <typename T>
+inline constexpr bool is_variant_type_v = get_variant_type_v<T> != Variant::VARIANT_MAX;
```

Comment:
Imo, this should just be
`inline constexpr bool inherits_object_v = TypeInherits<Object, T>::value;`
and I'd get rid of `get_variant_type_v`.

Reply:
Yeah, that makes sense. That comment was from earlier before I had a better understanding of what you were doing :)